### DISCUSSION TITLE
Trigger Sonos Voice commands/answers

### DISCUSSION THREAD

**Discussion by berndbarker** [hklages/node-red-contrib-sonos-plus#224] [Ideas]
At first, thanks a lot for all the the great stuff you’ve accomplished here!

Now an Idea:
There is this new Sonos Voice Service, which is handled fully local.
Is it possible to trigger voice answers via node-red-contrib-sonos-plus?

For example it would be great to trigger the voice command „hey sonos, what‘s on“ every time a playlist starts via node-red routine to get a nice audio feedback of the song and album now starting to play.

Thank you for looking into this!

**Comment by hklages** [comment]
> At first, thanks a lot for all the the great stuff you’ve accomplished here!

Thanks - always nice to hear :-)

This package uses the SONOS SOAP service. I could not find any new functionality making that possible. Give me some days to check with others. I use alexa and in case I would like to know that the command is "Alexa, was spielt gerade " (what is playing) -- but yes, that is not local.

Do you want to get a notification when a new playlist starts or when a new song in that playlist starts? 

You can do that already now - but it is some "work".  You need to install node-red-contrib-sonos-events to get a message when there is a title/album change. And you need to install a text to speech conversion such as polly.
1. Subscribe to media changes and get a message, when the title/album changes and store it in separate flow variables album, title
2. Anytime when that happen compare the data with the current values of that flow variables 
3. In case of a real change, trigger a notification with that title and album ([see here](https://github.com/hklages/node-red-contrib-sonos-plus/wiki/B.1-TTS-Notification)
You have to filter data in transition, etc.

**Reply by berndbarker** [reply]
Thank you for looking into it 👍 

The method you explained is the one I use right now. 
The real nice thing about using the integrated sonos voice service would be a much nicer and smoother voice, to be precise the voice of Giancarlo Esposito (Gustavo Fring in Breaking Bad, Moff Gideon in the Mandalorian, Stan Edgar in the Boys) and a much easier handling. 
The voice sounds pretty cool and natural, even when asking for „What’s playing“ and he answers that. 
It‘s all done offline, so I‘m really hoping that there is a way to trick the speaker to believe that the microphone just received a voice command. And just one voice enabled speaker in the setup would be enough, because if grouped, only one speaker has to have a microphone. The answer is played on each group member. So a group command followed by the voice command recognition triggered and an ungroup after would perfectly do the job, if you like to get a message on a speaker without built in sonos voice.
For me, the voice sounds really awesome, you have to check it out if you haven’t done yet :-)
I hope there’s a way, couldn’t believe these features aren’t going to be adressable via the api.
Thank you

**Comment by hklages** [comment]
I activated SONOS Voice on my SONOS Beam. Totally agree: amazing voice and quality, easy to use (but only available in Englsh language). 
What player do you use? It only works for voice enabled players. 

There are 2 api available. The "old" SOAP api (thats what I use in this package) and a new one, based on SONOS Cloud. The later is not easy to handle as it requires special authorization. Sonos seems to be in the transition from old SOAP api to the Cloud based. 

Next week I will have a deeper look into it.

**Reply by berndbarker** [reply]
I am using an Arc. Yes, only english available so far (I‘m german), but with this great voice I think I‘ll stay with that ;-)
Ah, okay, good to know. If the new api ist based on sonos cloud, does that mean it only works online?
Thank you big time for staying on this and have a deeper look into it, really appreciate it! Or just „Danke“ ;-)

**Comment by hklages** [comment]
I checked the SOAP Api and with to other guys: it is currently not possible. Hope that SONOS exposes the api in the future. 

Sorry.

**Reply by berndbarker** [reply]
That’s quite disappointing, but no reason at all for you to say sorry!

Thank you so much for all your research you’ve done and your answers here, let’s see what the (sonos) future brings! :-)
